designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Button.kt

```diff
@@ -66,15 +67,30 @@ fun Button(
 ) {
     when (style) {
         ButtonStyle.FILLED -> {
-            val textColor = if (enabled) TextColor.OnPrimary else TextColor.OnDisabledSurface
+            val textColor = if (enabled) {
+                when (colorStyle) {
+                    ColorStyle.DEFAULT -> TextColor.OnPrimary
+                    ColorStyle.ERROR -> TextColor.OnError
+                }
+            } else {
+                TextColor.OnDisabledSurface
+            }
+            val containerColor = when (colorStyle) {
+                ColorStyle.DEFAULT -> SurfaceColor.Primary
+                ColorStyle.ERROR -> SurfaceColor.Error
+            }
+            val contentColor = when (colorStyle) {
+                ColorStyle.DEFAULT -> TextColor.OnPrimary
+                ColorStyle.ERROR -> TextColor.OnError
+            }
```

Comment:
I feel like the cognitive complexity of this function is increased a lot now. It might be a good idea to break down the function into smaller functions and utility functions, so that we have byte size pieces to read and also avoid any code. duplications